### PR TITLE
Add Install section to DC/OS units

### PR DESCRIPTION
**Pull request by surdy**
## High-level description

What features does this change enable? What bugs does this change fix?
This enables the operator to enable a DC/OS systemd unit using `systemctl enable <unit-name>` after having disabled the unit using `systemctl disable <unit-name>` 

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-4479](https://jira.mesosphere.com/browse/DCOS_OSS-4479) Unable to enable DC/OS systemd units using "systemctl enable" after disabling

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-<number>](https://jira.mesosphere.com/browse/DCOS_OSS-<number>) Foo the Bar so it stops Bazzing.

## Checklist for all PRs

  - [X] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [X] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [X] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by surdy** [comment]
@branden @janisz please review. Since this is a change that touches all components, who else do you suggest I reach out for review.

**Comment by surdy** [comment]
@branden @janisz please review. Since this is a change that touches all components, who else do you suggest I reach out for review.

**Comment by surdy** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by janisz** [comment]
@gpaul Can you take a look? I'm not sure about this

**Comment by surdy** [comment]
@leehambley who on your team would be best person to be second reviewer for this ? Once I get 2 thumbs up I will bump-ee and update that branch as well.

**Comment by branden** [comment]
> Any reservations against me back-porting this to 1.12/1.11 ?

Not at all, and I appreciate your offer to do it! We'd want to backport to 1.10 as well, since it's still a supported version.

**Comment by surdy** [comment]
@mesosphere-mergebot bump-ee

**Comment by surdy** [comment]
@mesosphere-mergebot bump-ee

**Comment by janisz** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
sorry @surdy you've hit one of the most frustrating parts of our workflow. You need to rebase your branch on the latest master and since I don't have admin access to this repository I can't do it for you.

**Comment by leehambley** [comment]
I see that some people have commented and given a LGTM/ :thumbsup:, however this is at odds with our dc/os systemd rules:

- https://github.com/dcos/dcos/blob/master/docs/systemd-rules.md#rules-for-using-systemd-reliably

how should we resolve that?

**Comment by branden** [comment]
@leehambley That doc warns against using `Wants=`, `Before=`, etc. in component unit files, in order to prevent people from using them to get systemd to start components in a certain order (e.g. "start dcos-adminrouter before you start me"). Systemd units with complex dependency graphs can be a pain to operate, and systemd docs warn against using these directives.

All of our systemd units already have an implicit `WantedBy=` relationship to `dcos.target`, because we symlink their unit files into `dcos.target.wants/` during installation. That's how systemd knows that `dcos.target` is an alias for all of our various components. The problem this leaves is that a `systemctl disable` of a DC/OS unit will remove that symlink, destroying the only clue systemd has that these units are related. This PR duplicates that dependency information within unit files, so that `systemctl enable` can know where to restore the symlink.

**Comment by leehambley** [comment]
Responding to the mergebot 👋  - @surdy how would you like to proceed with this PR?

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@branden do you agree we can (could) merge this PR pair? If so, @surdy could update his branch and we can get this merged with some focus.

**Comment by gpaul** [comment]
The error is:
```
Mar 01 00:32:23 ip-10-10-0-194.us-west-2.compute.internal systemd[1]: dcos-diagnostics.service holdoff time over, scheduling restart.
Mar 01 00:32:23 ip-10-10-0-194.us-west-2.compute.internal systemd[1]: Cannot add dependency job for unit dcos-diagnostics.socket, ignoring: Unit is not loaded properly: Bad message.
```
...
```
Mar 01 00:32:18 ip-10-10-0-194.us-west-2.compute.internal dcos-diagnostics[11668]: time="2019-03-01T00:32:18Z" level=info msg="Start dcos-diagnostics"
Mar 01 00:32:18 ip-10-10-0-194.us-west-2.compute.internal dcos-diagnostics[11668]: time="2019-03-01T00:32:18Z" level=fatal msg="Unix socket not found"
Mar 01 00:32:18 ip-10-10-0-194.us-west-2.compute.internal systemd[1]: dcos-diagnostics.service: main process exited, code=exited, status=1/FAILURE
Mar 01 00:32:18 ip-10-10-0-194.us-west-2.compute.internal systemd[1]: Unit dcos-diagnostics.service entered failed state.
Mar 01 00:32:18 ip-10-10-0-194.us-west-2.compute.internal systemd[1]: dcos-diagnostics.service failed.
```

File: packages/dcos-diagnostics/extra/dcos-diagnostics.socket [hunk 1]
```diff
@@ -6,3 +6,6 @@ ListenStream=/run/dcos/dcos-diagnostics.sock
 SocketUser=root
 SocketGroup=dcos_adminrouter
 SocketMode=0660
+
+Install]
```
Comment (gpaul) [inline packages/dcos-diagnostics/extra/dcos-diagnostics.socket]: Ah yes - missing `[` here

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@surdy OK we're going to need some help from @orsenthil landing these changes.

**Comment by gpaul** [comment]
@surdy I recommend pushing your branch to the `mesosphere/dcos` repository then @janisz and I can add commits / rebase / push it over the finish line and avoid the RTT necessary to get it merged promptly. Alternatively, I'll just pull your commits into a new branch, rebase it, and create new PRs to merge this work. It's important work.

**Comment by makkes** [comment]
@surdy Can we expect any updates here in the near future? Otherwise we'd like to close this PR.

**Comment by surdy** [comment]
Sorry I have not been keeping up with this. I'm closing this and the related PRs for now.

**Pull request closed without merging**